### PR TITLE
refactor(Core/Player): created update player category

### PR DESCRIPTION
**Pull request by iraizo**
Follow up on https://github.com/azerothcore/azerothcore-wotlk/pull/6838
Based off my comment on https://github.com/azerothcore/azerothcore-wotlk/issues/6698

Moves around 2000 lines of code to PlayerUpdates.cpp, every function that starts or has something to do with Updating has been moved to there with its functions. 

This PR has been made at 4AM and my editor was a little broken so the `Player.cpp` changes will be added when i wake up.
Please give me some feedback on this change

Please add a WIP tag to this.

### CODE REVIEW

**Comment by iraizo** [comment]
Unfortunately the clangd extension also has clang-format inbuilt and I've just disabled the clang-format extension this is why Player.cpp is formatted by https://github.com/azerothcore/azerothcore-wotlk/pull/6848. I will either fix this or wait until that PR has been merged / a proper format has been made.

**Comment by iraizo** [comment]
welp wrong branch

**Comment by iraizo** [comment]
Okay fixed it

**Comment by FrancescoBorzi** [comment]
@iraizo there is a conflict in `Player.cpp`

**Comment by iraizo** [comment]
> @iraizo there is a conflict in `Player.cpp`

Working on it

**Comment by iraizo** [comment]
Well its gonna take a bit because all my editors crash trying to show the merge conflict or even opening the file, good thing were cleaning this up..

**Comment by iraizo** [comment]
https://github.com/azerothcore/azerothcore-wotlk/pull/6840/commits/a24338f03df42d4310069cbddaaf6eee3c5e3e18 Adds the changes made by https://github.com/azerothcore/azerothcore-wotlk/commit/65c9d3669f1ca714c66ac224a0aae9440f4bf3e6 So the merge is now properly done.

**Comment by FrancescoBorzi** [comment]
@iraizo you are missing some import:

```
src/server/game/Entities/Player/PlayerUpdates.cpp:605:23: fatal error: use of undeclared identifier 'PLAYER_SKILL_VALUE_INDEX'
  uint32 valueIndex = PLAYER_SKILL_VALUE_INDEX(itr->second.pos);
```

if you need help, just let me know

File: src/server/game/Entities/Player/PlayerUpdates.cpp [hunk 1]
```diff
@@ -0,0 +1,2048 @@
...
+#include "Vehicle.h"
+#include "WeatherMgr.h"
+
+// Zone Interval should be 1 second
+#define ZONE_UPDATE_INTERVAL (1 * IN_MILLISECONDS)
+
+#define CINEMATIC_UPDATEDIFF 500
+#define CINEMATIC_LOOKAHEAD (2 * IN_MILLISECONDS)
+
+void Player::Update(uint32 p_time) {
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/Player/PlayerUpdates.cpp:27]: the style is changed... the braces should be on new line to make it consistent with the rest of the proejct

**Comment by iraizo** [comment]
Weird, clangd didn't tell me that, I will fix it as soon as I can

File: src/server/game/Entities/Player/PlayerUpdates.cpp (hunk 1 above)
Comment (iraizo) [reply src/server/game/Entities/Player/PlayerUpdates.cpp:33]: Yeah this is why i marked it as WIP, please merge this after the clang format PR has been merged so both files are in the proper format

**Comment by iraizo** [comment]
@FrancescoBorzi Yeah i may require some help fixing up those errors when compiling, i dont have time at the moment to fix them myself.

**Comment by iraizo** [comment]
Closing due to merge errors, will create another PR

**Pull request closed without merging**